tests/backends/postgresql/test_server_side_cursors.py

```diff
@@ -31,17 +31,20 @@ def inspect_cursors(self):
 
     @contextmanager
     def override_db_setting(self, **kwargs):
-        for setting in kwargs:
-            original_value = connection.settings_dict.get(setting)
-            if setting in connection.settings_dict:
-                self.addCleanup(
-                    operator.setitem, connection.settings_dict, setting, original_value
-                )
+        cleanups = []
+        settings = connection.settings_dict
+
+        for name, value in kwargs.items():
+            if name in settings:
+                cleanups.append((operator.setitem, settings, name, settings[name]))
             else:
-                self.addCleanup(operator.delitem, connection.settings_dict, setting)
+                cleanups.append((operator.delitem, settings, name))
+            settings[name] = value
+
+        yield
 
-            connection.settings_dict[setting] = kwargs[setting]
-            yield
+        for operation, *args in reversed(cleanups):
+            operation(*args)
```

Comment:
Can we add a test to shows what was fixed?

Reply:
Done.